Dz_2 (scss)

ShittyWizard opened the pull request.

ShittyWizard committed:
Add styles and icons

ShittyWizard committed:
css -> scss
added mixin "placeholder"
added vars for colors and font-size

ShittyWizard committed:
edited font-size
todo-item_remove and todos-add_delete-all -> float: right

ShittyWizard committed:
move textarea.scss to todo-item.scss

ShittyWizard committed:
fix

ShittyWizard committed:
Добавил анимацию по hover для крестика.
Добавил "резиновость" через flexbox и float.

ShittyWizard committed:
Добавил 2 mixin : transition(аттрибут/свойство, время анимации) и priority-font(шрифт).
Добавил в переменные ширину и высоту "крестика" и записал отступы для него через "математику" над переменными.

ShittyWizard commented:
github pages -
https://shittywizard.github.io/public/index.html

но тут уже с js-ом.

DmitryMakhnev commented on src/styles/reset.scss:
```diff
@@ -0,0 +1,25 @@
+  margin-bottom: -20px;
+}
+
+.todos-main{
```
`reset.scss` для сброса стилей
а `.todos-title` и `.todos-main` должны быть вынесены в соответсвующие компоненты, см. разбор дз

DmitryMakhnev commented on src/styles/reset.scss:
```diff
@@ -0,0 +1,25 @@
+  font-weight: 100;
+  font-size: $fs_title;
+  color: $c-title;
+  margin-bottom: -20px;
```
Стили вида `margin-bottom: -20px;` не допустимы для описания компонента
Если нужно как-то встроить компонент в лайаут, нужно позиционировать от лайаута каскадом, см. MCSS

DmitryMakhnev commented on src/styles/project/todos-steps/todos-steps.scss:
```diff
@@ -0,0 +1,19 @@
+  background: $c-white;
+  border: solid 1px $c-border;
+  box-shadow: 0 3px 5px $c-shadow;
+  width: 99%;
```
`width: 99%;` и `width: 98%;` в данном кейсе не очень подходят, т.к. в зависимости от ширины контейнера отступы будут меняться.
В данном кейсе лучше использовать фиксированные боковые марджины (см. разбор дз)

DmitryMakhnev commented on src/styles/project/todos-add/todos-add.scss:
```diff
@@ -0,0 +1,57 @@
+  border-bottom: 1px solid $c-border;
+  box-shadow: 0 3px 5px $c-shadow;
+
+  display: flex;
```
Для данного кейса не нужно использовать флекс бокс

DmitryMakhnev commented on src/styles/project/todos-add/todos-add.scss:
```diff
@@ -0,0 +1,57 @@
+    @include priority-font('Helvetica Neue');
+    font-weight: 100;
+    font-size: $fs-large;
+    @include placeholder(Helvetica Neue);
```
Если посмотреть на 3 строчки выше, не консистентно

DmitryMakhnev commented on src/styles/config/mixins/placeholder.scss:
```diff
@@ -0,0 +1,13 @@
+@mixin placeholder($font) {
+    input::-webkit-input-placeholder {
+        font-family: $font, Arial, sans-serif;
```
А если цвет захочется поменять?
Или размер шрифта?
А если `font-width` не только 100?
Не надёжно.
миксин placeholder должен изолировать в себе проблему вендорных префиксов в псевдоклассах (см. разбор дз)

DmitryMakhnev commented on src/styles/config/mixins/priority-font.scss:
```diff
@@ -0,0 +1,3 @@
+@mixin priority-font($font){
+  font-family: $font, Arial, sans-serif;
```
Непонятный миксин.
Проблемы:
1) По всем стилям используется прямое название шрифта, что концептуально противоречит переменным, см разбор дз
2) Весь шрифт должен быть записан в переменную: $common-font: 'Helvetica Nune', Arial, sans-serif;

DmitryMakhnev commented on src/styles/project/todo-item/todo-item.scss:
```diff
@@ -0,0 +1,63 @@
+  .todo-item_mark {
+    float: left;
+    opacity: 0;
+    z-index: 2;
```
зачем `z-index`

DmitryMakhnev commented on public/index.html:
```diff
@@ -0,0 +1,71 @@
+
+		<div class="todos-list">
+			<div class="todo-item">
+				<input class="todo-item_mark" type="checkbox"  aria-label="Mark done">
```
Сейчас стили чекбокса описаны относительно компонентов `todo-item_mark`
См `input-checkbox` в разборе дз

DmitryMakhnev replied:
Нет исправления

DmitryMakhnev reviewed (commented):
Поправить комменты
Детально ознакомиться с разбором дз

ShittyWizard committed:
Исправления

Убрал todos-main и todos-title из reset.scss
Изменил width: 99% и width: 98% на margin'ы с обоих сторон ( на 2px и 4px соответственно)
Изменил миксины (добавил @content, не подумал об этом когда писал в первый раз)
Создал переменную $common-font: "Helvetica Neue", Arial, sans-serif;

DmitryMakhnev commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:ok-technopolis/preprocessor-stub.git"
```
Поправить согласно проекту

DmitryMakhnev commented on src/styles/base/button/button.scss:
```diff
@@ -0,0 +1,13 @@
+@import "../../config/globals";
+
+button {
```
Ни каких селекторов по тегу.
См. разбор дз

DmitryMakhnev commented on src/styles/base/checkbox/checkbox.scss:
```diff
@@ -0,0 +1,20 @@
+@import "../../config/globals";
+
+.todo-item_mark, .todo-item_mark-checkbox {
```
Идентично https://github.com/ok-technopolis/Shvedov_Oleg/pull/4#discussion_r115477715

DmitryMakhnev commented on src/styles/project/todo-item/todo-item.scss:
```diff
@@ -0,0 +1,64 @@
+    order: 10;
+  }
+
+  textarea {
```
А если будет ещё одна текстария в проекте?
Ни каких селекторов по тегу. Пишем тегу нужный класс. см. разбор дз

DmitryMakhnev commented on src/styles/reset.scss:
```diff
@@ -0,0 +1,11 @@
+@import "config/globals.scss";
+
+* {
```
См reset в разборе дз

DmitryMakhnev reviewed (commented):
Нет исправлений от прошлых замечаний.
Поправить всё с детальным разбором проблемных мест в сравнении с моим вариантом дз
